### ISSUE TITLE
[@angular-eslint/template/no-negated-async] Should no-negated-async also catch instances where the input variable to async pipe is negated?

### ISSUE THREAD

**Issue by miccou** [angular-eslint/angular-eslint#1566]
I came across some code in a PR where a developer had negated the actual input variable to the async pipe like this: `!var$ | async`. This caused some bugs and was caught, but I though that no-negated-async might have caught this issue. An observable cannot be negated like this.

Should no-negated-async catch this? Or should there be a new rule?

```JSON
{
  "rules": {
    "@angular-eslint/template/no-negated-async": "error"
  }
}
```

```HTML
<button [disabled]="!buttonDisabled$ | async">Click me!</button> <!-- this doesn't cause a lint error-->

<button [disabled]="!(buttonDisabled$ | async)">Click me!</button> <!-- this does cause a lint error-->
```

**Versions**

| package                                   | version |
| ----------------------------------        | ------- |
| `@angular-eslint/eslint-plugin-template`  | `14.0.2` |
| `@angular-eslint/template-parser`         | `14.0.2` |
| `@typescript-eslint/parser`               | `5.29.0` |
| `ESLint`                                  | `8.15.0` |
| `node`                                    | `18.15.0` |

- [x] I have tried restarting my IDE and the issue persists.
- [ ] I have updated to the latest supported version of the packages and checked my `ng version` output per the instructions given here.

**Comment by JamesHenry** [comment]
It's worth pointing out that syntactically we have no way of knowing that `buttonDisabled$` is an Observable. It's implied by the `$` but that is not enough of a guarantee.

Nevertheless, even if `buttonDisabled$` were a Promise to be used with the async pipe, the `!` still doesn't make sense as a use-case right?

I don't think there is any actual use-case where !VALUE with the async pipe makes sense?

So maybe we should just go ahead and flag it as part of the rule, despite it technically being a bit different?

It's very specific to the `async` pipe, and I don't think it's worth its own dedicated rule just for this...

**Comment by JamesHenry** [comment]
(Didn't mean to close then, sorry)

**Comment by abaran30** [comment]
Fair enough, and the additional check was simple enough to add to the rule; PR mentioned above.

**Comment by miccou** [comment]
Thanks for considering this request, I thought about your comments and I think it makes sense to extend this rule to something like "Ensures that async pipe results __and inputs__ are not negated. The rule then catches these async pipe antipatterns that can result in unexpected behaviour but aren't flagged by the IDE by default (that's what lint is all about!).

Really appreciate your work on this tooling!
